desktop/apps/auction_support/client/registration_form.coffee

```diff
@@ -103,18 +110,19 @@ module.exports = class RegistrationForm extends ErrorHandlingForm
       else
         resolve()
 
+  # Lock the form- action.finally() callback executes when form submission is complete regardless of success
   loadingLock: ($element, action) ->
     return if $element.hasClass('is-loading')
     $element.addClass 'is-loading'
-    action().finally => $element.removeClass 'is-loading'
+    action().finally => $element.removeClass 'is-loading' unless @comboForm
```

Comment:
Before JS promises were a standard spec of the language there were all sorts of features of promises provided by libraries [including `finally`](http://bluebirdjs.com/docs/api/finally.html). `finally` is similar to the `finally` in [try/catch/finally](https://www.w3schools.com/jsref/jsref_try_catch.asp) in that it runs some code after a promise resolves successful or not. That's why this will remove the spinner whether registration was successful or failed.

Fun note: With async/await these can be considered the same thing:

```javascript
fetch()
  .then(() => console.log('success!'))
  .catch((e) => console.log('fail!', e))
  .finally(() => console.log('moving on'))
```

```javascript
try {
  await fetch()
  console.log('success!')
} catch (e) {
  console.log('fail', e)
} finally {
  console.log('moving on')
}
```